pkg/manifestreader/common.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2020 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package manifestreader
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/cli-runtime/pkg/resource"
+	"k8s.io/kubectl/pkg/cmd/util"
+	"sigs.k8s.io/cli-utils/pkg/apply/solver"
+)
+
+type ManifestReader interface {
+	Read() ([]*resource.Info, error)
+}
+
+type ReaderOptions struct {
+	Factory          util.Factory
+	Validate         bool
+	Namespace        string
+	EnforceNamespace bool
+}
+
+func setNamespaces(factory util.Factory, infos []*resource.Info,
```

Comment:
This is a big function. Even though it's not exported, should we put a comment above it to summarize what it's doing?

Reply:
Done. I just missed copying the comments from the applier.